pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -921,17 +929,20 @@ func (o *operatingSystemConfig) calculateKeyForVersion(version int, worker *gard
 	if err != nil {
 		return "", err
 	}
+	kubeReserved := v1beta1helper.CalcluateEffectiveResourceReservations(o.values.KubeletConfig, worker.Kubernetes)
 
-	return CalculateKeyForVersion(version, kubernetesVersion, worker)
+	return CalculateKeyForVersion(version, kubernetesVersion, o.values.CredentialsRotationStatus, worker, o.values.NodeLocalDNSEnabled, kubeReserved)
 }
 
 // CalculateKeyForVersion is exposed for testing purposes only
 var CalculateKeyForVersion = calculateKeyForVersion
 
-func calculateKeyForVersion(oscVersion int, kubernetesVersion *semver.Version, worker *gardencorev1beta1.Worker) (string, error) {
+func calculateKeyForVersion(oscVersion int, kubernetesVersion *semver.Version, credentialsRotation *gardencorev1beta1.ShootCredentialsRotation, worker *gardencorev1beta1.Worker, nodeLocalDNSEnabled bool, kubeReserved *gardencorev1beta1.KubeletConfigReserved) (string, error) {
 	switch oscVersion {
 	case 1:
 		return KeyV1(worker.Name, kubernetesVersion, worker.CRI), nil
```

Comment:
Can you add a  TODO statement to remove this `KeyV1` when support for Kubernetes `1.30` is dropped?

Reply:
I've added a todo.